src/machine/mod.rs

```diff
@@ -75,7 +75,7 @@ pub trait SupportMachine: CoreMachine {
     fn set_running(&mut self, running: bool);
 
     // Erase all the states of the virtual machine.
-    fn reset(&mut self);
+    fn reset(&mut self, max_cycles: Option<u64>);
```

Comment:
The interface is confusing. Does `None` mean keeping the original max cycles or clear the max cycles limit?

Reply:
I have removed Option wrapper of max_cycles in the latest commit, please review it again